internal/agent/upgrade.go

```diff
@@ -117,24 +121,51 @@ func Upgrade(
 
 	utils.SetEnv(c.Env)
 
-	args := []string{"upgrade", "--system.uri", fmt.Sprintf("docker:%s", img)}
-	args = append(args,
-		"--auth-username", authUser,
-		"--auth-password", authPass,
-		"--auth-server-address", authServer,
-		"--auth-type", authType,
-		"--auth-registry-token", registryToken,
-		"--auth-identity-token", identityToken,
-	)
-
+	// Load the upgrade Config from the system
+	resetConfig, err := elementalConfig.ReadConfigRun("/etc/elemental")
+	if err != nil {
+		return err
+	}
 	if debug {
-		fmt.Printf("Running command: 'elemental %s'", strings.Join(args, " "))
+		resetConfig.Logger.SetLevel(log.DebugLevel)
+	}
+
+	// Generate an auth object
+	auth := &types.AuthConfig{
+		Username:      authUser,
+		Password:      authPass,
+		ServerAddress: authServer,
+		Auth:          authType,
+		IdentityToken: identityToken,
+		RegistryToken: registryToken,
+	}
+
+	// Override the default luet to pass the auth
+	// Remember to create the temp dir!
```

Comment:
Is this a TODO note? Who should remember it and when?

Reply:
Well this has history :D
Its just a note to mention that when creating a custom luet object you need to create a temp dir for it explicitely. Otherwise things go wrong very soon. I'll expand the comment a bit more!

Reply:
But there is nothing to do in this part of the code no? Maybe this reminder belongs closer to where the temp dir should be created.

Reply:
It should be created here, when you create the new luet. I hope to fix this in the future once the code is folder but currently if someone removes that line, unintended consequences will happen :D